paraatm/bdn/_bdn.py

```diff
@@ -0,0 +1,209 @@
+        model.add(Activation('relu'))
+        if self.odrop:
+            model.add(Dropout(self.odrop))
+        model.add(Dense(1,activation='linear',
+                        kernel_regularizer=l2(self.weight_decay),
+                        bias_regularizer=l2(self.weight_decay)))
+        optimizer_=Adam(self.lr)
+        model.compile(loss='mse',metrics=['mse'],optimizer=optimizer_)
+        return model
+    def _get_dnn_model(self):
+        """
+        Construct a deep neural network (dnn) model which is then passed to 'pred' method if flag_2 =='dnn'.
+        A dnn model is used to make predictions for input data which has linear or non-linear relationships with the output data, or target.
+        
+        
+        For more information about dnn model, please refer to:
+        https://en.wikipedia.org/wiki/Deep_learning#Deep_neural_networks
+        
+        Returns
+        -------
+        A sequential dnn model ready to be fit in 
+        """
+        if len(self.x_train.shape)!=2:
+            raise Exception('expect a 2 dimension array!')
+        in_shape = self.x_train.shape[-1]
+        model=Sequential()
+        model.add(Dense(self.num_unit,kernel_regularizer=l2(self.weight_decay),
+                       bias_regularizer=l2(self.weight_decay),
+                        kernel_initializer='random_uniform',
+                        input_dim=in_shape))
+        
+        model.add(Activation('relu'))
+        model.add(Dropout(self.idrop))
+        model.add(Dense(self.num_unit,kernel_regularizer=l2(self.weight_decay),
+                       bias_regularizer=l2(self.weight_decay),
+                        kernel_initializer='random_uniform',
+                       ))
+        model.add(Dropout(self.idrop))
+        model.add(Activation('relu'))  
+        if self.odrop:
+            model.add(Dropout(self.odrop))
+        model.add(Dense(1,activation='linear',
+                        kernel_regularizer=l2(self.weight_decay),
+                        bias_regularizer=l2(self.weight_decay)))
+        optimizer_=Adam(self.lr)
+        model.compile(loss='mse',metrics=['mse'],optimizer=optimizer_)
+        return model        
+        
+
+    def pred(self,flag_1 = True,flag_2 = 'rnn',custom_model = None):
+        """
+        pred is used to generate results for the deep learning model
+        
+        Parameters
+        ----------
+        flag_1: Boolen
+            By default: it's set as 'True'
+            flag_1 == True: prediction with uncertainty
+            flag_1 == False: deterministic result
+        
+        flag_2: str
+            By default, it's set as 'rnn'
+            flag_2 == 'rnn': generate results for the embedded rnn model
+            flag_2 == 'dnn': generate results for the embedded dnn model
+            flag_2 == 'custom': generate results for the customized model
+            When flag_2 == 'custom', a customized model must be provided
```

Comment:
Can you use more descriptive names for these?  Instead of `flag_1` and `flag_2`, perhaps `get_stdev` (or `get_var`) and `model_type`?

Reply:
Hi John,

Thanks for the comments. ' flag_1' now is changed to 'pred_type'. 'flag_2' is changed to 'model_type'.

Reply:
Two comments:
* Why did you move `pred_type` to be an argument to `__init__` instead of `pred`?  I think it makes more sense to be an argument of `pred`.  That way, the user has the flexibility to get different types of predictions after creating the model.
* The variable name is not consistent with its type.  `True` and `False` are not types of predictions.  It would be more logical to name the variable in a way that is consistent with the `True` and `False` values.  That is why my original suggestion was `get_stdev` or `get_var` (get standard deviation or variance).  This is more intuitive because it is clear that `get_stdev=True` means "yes, I want to also retrieve the standard deviation for the prediction".  It is not intuitive why `pred_type=True` would mean that the result should include uncertainty.  Other options would be `return_stdev`, `return_var`, `return_uncertainty`, etc.